### PR TITLE
Use relax rules control when performing updates

### PR DESCRIPTION
**Pull request by soisik**
See #165

### CODE REVIEW

**Comment by coudot** [comment]
Ok to add it to 2.1.6. As the new parameter is optional it should not break existing configurations

File: src/main/java/org/lsc/jndi/JndiServices.java [hunk 1]
```diff
@@ -903,6 +929,15 @@ private boolean doApply(final JndiModifications jm) throws CommunicationExceptio
 			
 			return false;
 		}
+		finally {
+			// clear requestControls for future use of the JNDI context
+			try {
+				ctx.setRequestControls(null);
```
Comment (rouazana) [inline src/main/java/org/lsc/jndi/JndiServices.java:935]: Do you think this configuration of the LdapContext is thread-safe? It seems ctx is shared by all threads using JndiServices, no?

**Comment by remk** [comment]
Do you think you can add a test for this ?

File: src/main/java/org/lsc/jndi/JndiServices.java (hunk 1 above)
Comment (soisik) [reply src/main/java/org/lsc/jndi/JndiServices.java]: you're right, I changed the way we use this control to derive the context for updates so the initial/shared context is not impacted by the control.

**Comment by soisik** [comment]
@remk : Unit test added, but since community version of OpenDJ doesn't support this control (the EE does), it's a bit limited; it just checks that the control is added or not depending on the context called.
Since it was impacted, I wrote an extra test for recursive delete as it didn't exists.

**Comment by coudot** [comment]
@remk @rouazana @soisik if all is ok for you, can we merge this for 2.1.6?

**Review by rouazana — APPROVED** [review]

**Pull request merged**